vm/devices/storage/disk_nvme/nvme_driver/fuzz/fuzz_main.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#![cfg_attr(all(target_os = "linux", target_env = "gnu"), no_main)]
+
+//! A 2-way fuzzer developed to fuzz the nvme driver from the Guest side with arbitrary driver
+//! action and from the Host side with arbitrary responses from the backend.
+mod fuzz_emulated_device;
+mod fuzz_nvme_driver;
+
+use crate::fuzz_nvme_driver::FuzzNvmeDriver;
+
+use arbitrary::Unstructured;
+use lazy_static::lazy_static;
+use pal_async::DefaultPool;
+use std::sync::Mutex;
+use xtask_fuzz::fuzz_target;
+
+// Input bytes we want to use
+const MIN_INPUT:usize=200;
+
+// Use lazy_static to allow swapping out underlying vector
+lazy_static! {
+    pub static ref RAW_DATA: Mutex<Vec<u8>> = Mutex::new(Vec::new());
+}
+
+/// Uses static RAW_DATA to generate a vector of len=num_bytes with arbitrary bytes
+pub fn get_raw_data(num_bytes: usize) -> arbitrary::Result<Vec<u8>>{
+    // Lock RAW_DATA before consuming
+    let mut raw_data = RAW_DATA.lock().unwrap();
+
+    // Case: Not enough bytes, return Error
+    if raw_data.len() < num_bytes {
+        return Err(arbitrary::Error::NotEnoughData);
+    }
+
+    // Consume bytes from RAW_DATA
+    let drained: Vec<u8> = raw_data.drain(0..num_bytes).collect();
```

Comment:
This could probably use split_off instead of drain?

Reply:
For a split at X, the [split_off](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.split_off) function updates the existing vector to [0..X) and returns a vector [X..len). If we consume the returned vector, it would mean consuming data from end -> start of the input arbitrary vector. 
This approach will still be deterministic and I think it should be fine .. any concerns with consuming backwards like this? [Trying to avoid doing a weird switcheroo after split_off]

Reply:
Oh good catch. I don't think that's a problem though. Like you said it's still deterministic.

Reply:
Now I am using split_off instead of drain. closing this